src/bluesearch/entrypoint/database/parse.py

```diff
@@ -14,49 +32,72 @@
 def get_parser() -> argparse.ArgumentParser:
     """Create a parser."""
     parser = argparse.ArgumentParser(
-        description="Parse article.",
+        description="Parse one or several articles.",
     )
     parser.add_argument(
         "article_type",
         type=str,
         choices=("cord19-json", "pmc-xml"),
-        help="""Article source type.""",
+        help="""
+        Article format. If parsing several articles, all articles
+        must have same format.
+        """,
     )
     parser.add_argument(
         "input_path",
-        type=str,
-        help="""Path to the file/directory to be parsed.""",
+        type=Path,
+        help="""
+        Path to a file or directory. If a directory, all articles
+        inside the directory will be parsed.
+        """,
     )
     parser.add_argument(
         "output_path",
-        type=str,
-        help="""Path where the parsed article is saved.""",
+        type=Path,
+        help="""
+        Path to a directory where parsed article(s) will be saved.
+        If it does not exist yet, a directory with this path is created.
+        """,
     )
     return parser
 
 
 def run(
     *,
     article_type: str,
-    input_path: str,
-    output_path: str,
+    input_path: Path,
+    output_path: Path,
 ) -> None:
-    """Parse an article.
+    """Parse one or several articles.
 
     Parameter description and potential defaults are documented inside of the
     `get_parser` function.
     """
-    parser_inst: ArticleParser
-    if article_type == "cord19-json":
-        with open(input_path) as f_input:
-            parser_inst = CORD19ArticleParser(json.load(f_input))
-    elif article_type == "pmc-xml":
-        parser_inst = PubmedXMLParser(input_path)
-
+    inputs: Iterable[Path]
+    if input_path.is_file():
+        inputs = [input_path]
+    elif input_path.is_dir():
+        inputs = input_path.glob("*")
     else:
-        raise ValueError(f"Unsupported article type {article_type}")
+        raise ValueError(
+            "Argument 'input_path' should be a path to an existing file or directory!"
+        )
+
+    for inp in inputs:
+        parser_inst: ArticleParser
+        if article_type == "cord19-json":
+            with inp.open("r") as f_inp:
+                parser_inst = CORD19ArticleParser(json.load(f_inp))
+        elif article_type == "pmc-xml":
+            parser_inst = PubmedXMLParser(inp)
+        else:
+            raise ValueError(f"Unsupported article type {article_type}")
+
+        article = Article.parse(parser_inst)
 
-    article = Article.parse(parser_inst)
+        output_path.mkdir(exist_ok=True)
+        # If we used .with_suffix(), then we would get file.pdf.xml --> file.pkl
```

Comment:
I'm getting a different result 🤔 
```python
>>> p = pathlib.Path("a.pdf.xml")
>>> p.with_suffix(".pkl")
PosixPath('a.pdf.pkl')
```

Reply:
I was referring to the following line, where `.stem` is used. 

I did not k(now until I saw a test failing) that `.stem` removes _all_ extensions from a given file path. I think this is not desirable, since the previous extensions could actually carry information on the article, and would help identifying the source in a unique way.

The following snipped hopefully clarifies what I mean.
```python
>>> fname = Path("neuroscience_article.xml.json")
>>> out_dir = Path("out_pkls/")
>>> (out_dir / fname.stem).with_suffix(".pkl")
PosixPath('out_pkls/neuroscience_article.pkl')
>>> out_dir / (fname.stem + ".pkl")
PosixPath('out_pkls/neuroscience_article.xml.pkl')
```

Reply:
I'm still not sure `Path.stem` removes all extensions:
```python
>>> fname = Path("neuroscience_article.xml.json")
>>> fname.stem
'neuroscience_article.xml'
```
I think it's just because in your example `(out_dir / fname.stem) = "out_pkls/neuroscience_article.xml"` and the subsequent `with_suffix` replaces `.xml` by `.pkl`.

For example this would work:
```python
>>> inp
PosixPath('/some/dir/neuroscience_article.xml.json')
>>> out_dir
PosixPath('out_pkls')
>>> out_dir / inp.with_suffix(".pkl").name
PosixPath('out_pkls/neuroscience_article.xml.pkl')
```

Reply:
You're right! Thanks for the clarification, fixed in b6bdb08.